test/fixtures/es-module-loaders/transform-source.mjs

```diff
@@ -1,14 +0,0 @@
-export async function transformSource(
-  source, { url, format }, defaultTransformSource) {
-  if (format === 'module') {
-    if (typeof source !== 'string') {
-      source = new TextDecoder().decode(source);
-    }
-    return {
-      source: source.replace(`'A message';`, `'A message'.toUpperCase();`)
-    };
-  } else { // source could be a buffer, e.g. for WASM
-    return defaultTransformSource(
-      source, {url, format}, defaultTransformSource);
-  }
-}
```

Comment:
Same as above.

Reply:
Agree with the above but I think not this because transform source no-longer exists. It's incorporated into the `load` hook; I can't think of a reason for adding a specific case to the load-hook test where it handles a transformation because of how the hook's purpose/behaviour: whatever you do between invocation and output is your business and it doesn't care (and has basically no power to stop you) as long as your output is intelligible.

I think the [`import('emsHook/virtual.mjs')` case](https://github.com/nodejs/node/pull/37468/files/e7422eb13ca4df47df66e86803c3e417065c067c#diff-4a8a0345f25cf8762e6ffd591dea32050b6b51dc0e02638247ec10595fe19f29R47) effectively covers this because it provides an arbitrary `source` return value (which could have come from babel, coffeescrript, tsc, or whatever); whether it did or didn't doesn't matter (the 3 possible outcomes are the same: valid `source`, invalid `source`, an error thrown).

Reply:
The equivalent I guess would be returning ` source` where that `source` is a function of applying `defaultLoad` and using that source value returned.

The overall concept of testing chaining a transform with another source is an important one to retain I think, and will be useful to extend when the full chaining applies.

Reply:
There's [a case](https://github.com/nodejs/node/pull/37468/files/e7422eb13ca4df47df66e86803c3e417065c067c#diff-3a607d0f76bd2ac5c7703223f1e4413cde25cb367c1944da51099cc54117fbd9R10) for that :)